miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java

```diff
@@ -0,0 +1,550 @@
+    authorizationManager.throwIfNonAdminOrMatchingOwner(securityManager.getUserByLoginName(deleteQc.getQcCreator()));
+    managed.getLibraryQCs().remove(deleteQc);
+    libraryQcDao.remove(deleteQc);
+    libraryDao.save(managed);
+  }
+
+  /**
+   * Loads persisted objects into library fields. Should be called before saving libraries. Loads all member objects <b>except</b>
+   * <ul>
+   * <li>creator/lastModifier User objects</li>
+   * </ul>
+   * 
+   * @param library the Library to load entities into. Must contain at least the IDs of objects to load (e.g. to load the persisted Sample
+   *          into library.sample, library.sample.id must be set)
+   * @throws IOException
+   */
+  private void loadChildEntities(Library library) throws IOException {
+    if (library.getSample() != null) {
+      library.setSample(sampleDao.get(library.getSample().getId()));
+    }
+    if (library.getLibraryType() != null) {
+      library.setLibraryType(getLibraryTypeById(library.getLibraryType().getId()));
+    }
+    if (library.getLibrarySelectionType() != null) {
+      library.setLibrarySelectionType(getLibrarySelectionTypeById(library.getLibrarySelectionType().getId()));
+    }
+    if (library.getLibraryStrategyType() != null) {
+      library.setLibraryStrategyType(getLibraryStrategyTypeById(library.getLibraryStrategyType().getId()));
+    }
+    List<Index> managedIndices = new ArrayList<>();
+    for (Index index : library.getIndices()) {
+      Index managedIndex = indexDao.getIndexById(index.getId());
+      if (managedIndex != null) managedIndices.add(managedIndex);
+    }
+    library.setIndices(managedIndices);
+    if (isDetailedLibrary(library)) {
+      DetailedLibrary lai = (DetailedLibrary) library;
+      if (lai.getKitDescriptor() != null) {
+        lai.setKitDescriptor(kitDescriptorDao.getKitDescriptorById(lai.getKitDescriptor().getId()));
+      }
+      if (lai.getLibraryDesign() != null) {
+        lai.setLibraryDesign(libraryDesignDao.getLibraryDesign(lai.getLibraryDesign().getId()));
+      }
+      if (lai.getLibraryDesignCode() != null) {
+        lai.setLibraryDesignCode(libraryDesignCodeDao.getLibraryDesignCode(lai.getLibraryDesignCode().getId()));
+      }
+      validateLibraryDesignValues(library);
+    }
+  }
+
+  /**
+   * Updates all timestamps and user data associated with the change
+   * 
+   * @param library the Library to update
+   * @throws IOException
+   */
+  private void setChangeDetails(Library library) throws IOException {
+    User user = authorizationManager.getCurrentUser();
+    library.setLastModifier(user);
+  }
+
+  /**
+   * Copies modifiable fields from the source Library into the target Library to be persisted
+   * 
+   * @param target the persisted Library to modify
+   * @param source the modified Library to copy modifiable fields from
+   * @throws IOException
+   */
+  private void applyChanges(Library target, Library source) throws IOException {
```

Comment:
Aha, I had missed the part where the things from `loadChildEntities` first need to have (possibly) changed IDs set here.